src/xdp/program.c

```diff
@@ -1407,11 +1438,16 @@ XdpProgramDetachRxQueue(
         //
         RemoveEntryList(&ProgramBinding->RxQueueEntry);
         InitializeListHead(&ProgramBinding->RxQueueEntry);
-
-        XdpProgramCompileNewProgram(RxQueue, &CompiledProgram);
         
         XdpRxQueueDeregisterNotifications(RxQueue, &ProgramBinding->RxQueueNotificationEntry);
-        XdpRxQueueSetProgram(RxQueue, CompiledProgram, NULL, NULL);
+        if (IsListEmpty(XdpRxQueueGetProgramBindingList(ProgramBinding->RxQueue))) {
+            XdpRxQueueSetProgram(RxQueue, NULL, NULL, NULL);
+        } else {
+            //
+            // Update the program in-place because we are down sizing the program bindings.
+            //
+            XdpRxQueueSync(RxQueue, XdpProgramCompileUpdateProgram, RxQueue);
```

Comment:
Ideally, we'd avoid doing this compile on the data path, since it adds latency, but I think simplicity here is the way to go.